lsp/server/src/diagnostic/DiagnosticSettings.ts

```diff
@@ -0,0 +1,66 @@
+/*
+ * Copyright (c) 2024, salesforce.com, inc.
+ * All rights reserved.
+ * SPDX-License-Identifier: MIT
+ * For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/MIT
+ */
+
+export type ProducerId = 
+    | 'misspelled-uiapi' 
+    | 'adapters-local-change-not-aware'; 
+
+/**
+ * data structure for diagnostic suppression configuration.
+ */
+export type DiagnosticSettings = {
+    maxNumberOfProblems: number;
+    suppressAll?: boolean;
+    suppressedIds?: Set<ProducerId>
+}
+
+/**
+ * Check if the individual diagnostic should be suppressed.
+ * @param settings The suppression config
+ * @param id The configuration id to check.
+ * @returns True if suppressed.
+ */
+export function isTheDiagnosticSuppressed(settings: DiagnosticSettings, id: ProducerId) {
+    return settings.suppressAll === true || settings.suppressedIds?.has(id);
+}
+
+export const defaultDiagnosticSettings: DiagnosticSettings = {
+    suppressAll: false,
+    suppressedIds: new Set(),
+    maxNumberOfProblems: 200,
+}
+
+/**
+ * Take in currentSetting and an input, return a new diagnosticSettings
+ * @param currentSetting the current setting
+ * @param input the input, a well format tree is like 
+ * 
+ *  {
+            "suppressAll": true,
+            "suppressedIds": ["adapters-local-change-not-aware"],
+            "maxProblemNumber": 50
+    }
+ * 
+ * @returns the settings for diagnostics
+ */
+export function getSettings(currentSetting: DiagnosticSettings, input: any): DiagnosticSettings {
```

Comment:
Is it better to have only 1 param `input`? All missing parts can be picked up from `defaultDiagnosticSettings`

Reply:
sure, I over thought about the setting system, let me simplify it.

Reply:
done